src/main/java/de/eydamos/backpack/misc/BackpackUsageCache.java

```diff
@@ -0,0 +1,190 @@
+package de.eydamos.backpack.misc;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+import de.eydamos.backpack.Backpack;
+import de.eydamos.backpack.network.message.MessageBackpackInfoRequest;
+
+@SideOnly(Side.CLIENT)
+public class BackpackUsageCache {
+
+    private static final Map<String, CacheEntry> cache = new ConcurrentHashMap<>();
+    private static final Map<String, Long> requestTimestamps = new ConcurrentHashMap<>();
+
+    // Configuration constants
+    private static final long CACHE_DURATION_MS = 60000; // 1 minute
+    private static final long REQUEST_THROTTLE_MS = 1000; // 1 second throttle between requests
+    private static final int MAX_CACHE_SIZE = 100; // Prevent memory leaks
+
+    private static final ReadWriteLock cacheLock = new ReentrantReadWriteLock();
+
+    private static class CacheEntry {
+
+        public final BackpackSlotUsageInfo info;
+        public final long timestamp;
+        public final boolean isValid;
```

Comment:
Why does the entry have a valid flag? imo, the methods that invalidate cache entries should determine if it is valid or not, and evict the entry instantly